toplevel/toploop.ml

```diff
@@ -18,11 +18,14 @@ include Topcommon
 include Topeval
 
 (* Read and execute commands from a file, or from stdin if [name] is "". *)
+type input_type =
+  | Stdin
+  | File of string
+  | String of string
 
 let use_print_results = ref true
 
-let use_channel ppf ~wrap_in_module ic name filename =
-  let lb = Lexing.from_channel ic in
+let use_channel ppf ~wrap_in_module lb name filename =
```

Comment:
maybe this should be renamed into `use_lexbuf`?

Reply:
Note: two lines below there is a call to `Location.init lb filname`. I think that it would be nicer if the producer of the lexbuf was in charge of configuration its location correctly, because it has more information on how to set this. However keep things as-is is less invasive and it sort of works with the requirement to have `filename` anyway for other purposes, so I propose to leave things as-is. (There are other parts of the existing codebase that are ugly, and it's just how it is.)

Reply:
renamed to use_lexbuf